### PR TITLE
Colo: Refactor remote command dispatch for create and delete

### PR DESCRIPTION
**Pull request by t-nelson**
#### Problem

`net/colo.sh create` and `net/colo.sh delete` remote script dispatch uses a precarious `eval`/`heredoc` combo on an external template that demands strategic escaping to control whether variables are expanded locally or remotely.  This precludes the templates from being submitted to `shellcheck`

#### Summary of Changes

- Move local variable expansions into the calling `heredoc`, allowing...
  - the script templates to be appended without `eval`, allowing...
    - the insane escaping mess to be eliminated, allowing...
      - renaming templates so they're picked up by `shellcheck`
- Fix `shellcheck` complaints in templates

### CODE REVIEW

File: net/scripts/colo-node-onacquire.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+if [ ! -f "$SOLANA_LOCK_FILE" ]; then
+  exec 9>>"$SOLANA_LOCK_FILE"
+  flock -x -n 9 || ( echo "Failed to acquire lock!" 1>&2 && exit 1 )
+  ( [ -n "$SOLANA_USER" ] && {
+    echo "export SOLANA_LOCK_USER=$SOLANA_USER"
+    echo "export SOLANA_LOCK_INSTANCENAME=$INSTANCE_NAME"
+    echo "[ -v SSH_TTY -a -f \"${HOME}/.solana-motd\" ] && cat \"${HOME}/.solana-motd\" 1>&2"
+  } >&9 ) || ( rm "$SOLANA_LOCK_FILE" && echo "SOLANA_USER undefined" 1>&2 && false )
+  exec 9>&-
+  cat > /solana-scratch/id_ecdsa <<EOK
```
Comment (mvines) [inline net/scripts/colo-node-onacquire.sh]: EOK/EOAK/EOM, is there any significance to the different words used here?  If no I'd prefer just a consistent `EOF` (or whatever), so my brain doesn't need to wonder why they aren't all the same

File: net/scripts/colo-node-onacquire.sh [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+${NETWORK_INFO}
+${CREATION_INFO}
+EOM
+
+  # Stamp creation MUST be last!
+  touch /solana-scratch/.instance-startup-complete
+else
+  # shellcheck disable=SC1090
+  exec 9<"$SOLANA_LOCK_FILE" && flock -s 9 && . "$SOLANA_LOCK_FILE" && exec 9>&-
+  echo "${INSTANCE_NAME} candidate is already ${SOLANA_LOCK_INSTANCENAME}" 1>&2
```
Comment (mvines) [inline net/scripts/colo-node-onacquire.sh]: The mixing of $FOO and ${BAR} is a little perturbing.  Pick one style and use it consistently.  I once used ${BAR} all the time, but these days I prefer $FOO to reduce syntax except in situations that require the additional curlies to disambiguate: `${FOO}BAR`

File: net/scripts/colo-node-onacquire.sh [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+CREATION_INFO="${CREATION_INFO:-"Creation info unavailable"}"
+
+if [ ! -f "$SOLANA_LOCK_FILE" ]; then
+  exec 9>>"$SOLANA_LOCK_FILE"
+  flock -x -n 9 || ( echo "Failed to acquire lock!" 1>&2 && exit 1 )
+  ( [ -n "$SOLANA_USER" ] && {
+    echo "export SOLANA_LOCK_USER=$SOLANA_USER"
+    echo "export SOLANA_LOCK_INSTANCENAME=$INSTANCE_NAME"
+    echo "[ -v SSH_TTY -a -f \"${HOME}/.solana-motd\" ] && cat \"${HOME}/.solana-motd\" 1>&2"
+  } >&9 ) || ( rm "$SOLANA_LOCK_FILE" && echo "SOLANA_USER undefined" 1>&2 && false )
```
[the remark this answers is not in the document]
Comment (t-nelson) [reply net/scripts/colo-node-onacquire.sh]: I'll see what I can do

File: net/scripts/colo-node-onacquire.sh (hunk 1 above)
Comment (t-nelson) [reply net/scripts/colo-node-onacquire.sh]: Key/AuthorizedKeys/MOTD, respectively.  At one point this block was inside an `<<EOF`, hence the divergence.  They can be renamed now

File: net/scripts/colo-node-onacquire.sh [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+# These variable must be set before the main body is called
+SOLANA_LOCK_FILE="${SOLANA_LOCK_FILE:?}"
+INSTANCE_NAME="${INSTANCE_NAME:?}"
+SSH_AUTHORIZED_KEYS="${SSH_AUTHORIZED_KEYS:?}"
+SSH_PRIVATE_KEY_TEXT="${SSH_PRIVATE_KEY_TEXT:?}"
+SSH_PUBLIC_KEY_TEXT="${SSH_PUBLIC_KEY_TEXT:?}"
+NETWORK_INFO="${NETWORK_INFO:-"Network info unavailable"}"
+CREATION_INFO="${CREATION_INFO:-"Creation info unavailable"}"
+
+if [ ! -f "$SOLANA_LOCK_FILE" ]; then
```
[the remark this answers is not in the document]
Comment (danpaul000) [reply net/scripts/colo-node-onacquire.sh]: No argument, but why is that your preference?

[the remark this answers is not in the document]
Comment (mvines) [reply net/scripts/colo-node-onacquire.sh]: To keep it in the shell mostly.   `[[` also supports some regex features that `[` doesn't.

**Pull request merged**